nssp_demo/collate_score_tables.R

```diff
@@ -0,0 +1,212 @@
+
+  dirs <- tibble::tibble(
+    dir_path =
+      fs::dir_ls(dir_of_forecast_date_dirs,
+        type = "directory"
+      )
+  ) |>
+    dplyr::mutate(dir_name = fs::path_file(dir_path)) |>
+    dplyr::filter(purrr::map_lgl(dir_name, matches_diseases)) |>
+    dplyr::pull(dir_path)
+
+  return(dirs)
+}
+
+process_loc_date_score_table <- function(model_run_dir) {
+  table_path <- fs::path(model_run_dir,
+    "score_table",
+    ext = "rds"
+  )
+  location <- fs::path_file(model_run_dir)
+  if (!(fs::file_exists(table_path))) {
+    warning(glue::glue(
+      "No `score_table.rds` found for location ",
+      "{location} in directory {model_run_dir}"
+    ))
+    return(NULL)
+  }
+  score_table <- readRDS(table_path)
+  score_table$quantile_scores$location <- location
+  score_table$sample_scores$location <- location
+  return(score_table)
+}
+
+process_date_score_table <- function(date_fit_dir) {
+  table_path <- fs::path(date_fit_dir,
+    "score_table",
+    ext = "rds"
+  )
+  table_dir <- fs::path_file(date_fit_dir)
+
+  report_date <- stringr::str_match(
+    table_dir,
+    "r_(([0-9]|-)+)_f"
+  )[2] |>
+    lubridate::ymd()
+
+  score_table <- readRDS(table_path)
+  score_table$quantile_scores$report_date <- report_date
+  score_table$sample_scores$report_date <- report_date
+  return(score_table)
+}
+
+bind_tables <- function(list_of_table_pairs) {
+  sample_metrics <- purrr::map(
+    list_of_table_pairs,
+    \(x) {
+      attr(x$sample_scores, "metrics")
+    }
+  ) |>
+    unlist() |>
+    unique()
+  quantile_metrics <- purrr::map(
+    list_of_table_pairs,
+    \(x) {
+      attr(x$quantile_scores, "metrics")
+    }
+  ) |>
+    unlist() |>
+    unique()
+
+  sample_scores <- purrr::map(
+    list_of_table_pairs,
+    \(x) {
+      x$sample_scores
+    }
+  ) |>
+    data.table::rbindlist(fill = TRUE)
+
+  quantile_scores <- purrr::map(
+    list_of_table_pairs,
+    \(x) {
+      x$quantile_scores
+    }
+  ) |>
+    data.table::rbindlist(fill = TRUE)
+
+  attr(sample_scores, "metrics") <- sample_metrics
+  attr(quantile_scores, "metrics") <- quantile_metrics
+
+
+  return(list(
+    sample_scores = sample_scores,
+    quantile_scores = quantile_scores
+  ))
+}
+
+collate_scores_for_date <- function(model_run_dir,
+                                    score_file_name = "score_table",
+                                    score_file_ext = "rds",
+                                    save = FALSE) {
+  message(glue::glue("Processing scores from {model_run_dir}..."))
+  locations_to_process <- fs::dir_ls(model_run_dir,
+    type = "directory"
+  )
+  location_score_tables <- purrr::map(
+    locations_to_process,
+    process_loc_date_score_table
+  )
+
+  date_score_table <- bind_tables(location_score_tables)
+
+  if (save) {
+    save_path <- fs::path(model_run_dir,
+      score_file_name,
+      ext = score_file_ext
+    )
+    message(glue::glue("Saving score table to {save_path}..."))
+    saveRDS(date_score_table, save_path)
```

Comment:
consider [`write_rds`](https://readr.tidyverse.org/reference/read_rds.html)

Reply:
Done in  52f0033

Reply:
Doesn't appear to be done

Reply:
actually done in 44519e1